cpp/src/neighbors/detail/cagra/compute_distance-ext.cuh

```diff
@@ -0,0 +1,739 @@
+  vpq_descriptor_spec<DistanceType::L2Expanded, 16, 256, 8, 2, half, uint8_t, uint32_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 16, 256, 8, 4, half, uint8_t, uint32_t, float>,
+  standard_descriptor_spec<DistanceType::L2Expanded, 32, 512, uint8_t, uint32_t, float>,
+  standard_descriptor_spec<DistanceType::InnerProduct, 32, 512, uint8_t, uint32_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 32, 512, 8, 2, half, uint8_t, uint32_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 32, 512, 8, 4, half, uint8_t, uint32_t, float>,
+  standard_descriptor_spec<DistanceType::L2Expanded, 8, 128, float, uint64_t, float>,
+  standard_descriptor_spec<DistanceType::InnerProduct, 8, 128, float, uint64_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 8, 128, 8, 2, half, float, uint64_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 8, 128, 8, 4, half, float, uint64_t, float>,
+  standard_descriptor_spec<DistanceType::L2Expanded, 16, 256, float, uint64_t, float>,
+  standard_descriptor_spec<DistanceType::InnerProduct, 16, 256, float, uint64_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 16, 256, 8, 2, half, float, uint64_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 16, 256, 8, 4, half, float, uint64_t, float>,
+  standard_descriptor_spec<DistanceType::L2Expanded, 32, 512, float, uint64_t, float>,
+  standard_descriptor_spec<DistanceType::InnerProduct, 32, 512, float, uint64_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 32, 512, 8, 2, half, float, uint64_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 32, 512, 8, 4, half, float, uint64_t, float>,
+  standard_descriptor_spec<DistanceType::L2Expanded, 8, 128, half, uint64_t, float>,
+  standard_descriptor_spec<DistanceType::InnerProduct, 8, 128, half, uint64_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 8, 128, 8, 2, half, half, uint64_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 8, 128, 8, 4, half, half, uint64_t, float>,
+  standard_descriptor_spec<DistanceType::L2Expanded, 16, 256, half, uint64_t, float>,
+  standard_descriptor_spec<DistanceType::InnerProduct, 16, 256, half, uint64_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 16, 256, 8, 2, half, half, uint64_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 16, 256, 8, 4, half, half, uint64_t, float>,
+  standard_descriptor_spec<DistanceType::L2Expanded, 32, 512, half, uint64_t, float>,
+  standard_descriptor_spec<DistanceType::InnerProduct, 32, 512, half, uint64_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 32, 512, 8, 2, half, half, uint64_t, float>,
+  vpq_descriptor_spec<DistanceType::L2Expanded, 32, 512, 8, 4, half, half, uint64_t, float>>;
+
+template <typename DataT, typename IndexT, typename DistanceT, typename DatasetT>
+auto dataset_descriptor_init(const cagra::search_params& params,
```

Comment:
This one is fully generated file, so it would be not ideal to push the docs through python layer. I'll look to add these explanation docstrings elsewhere